### ISSUE TITLE
Revision request

### ISSUE THREAD

**Issue by deargle** [deargle-classes/msbx5500-spring-2020-project#9]
*	Business Understanding
    *	looks good
*	Data Understanding
    *	This section should say which ctu-13 scenario was used (the first one, with a link to it).
    *	description of the use of kddcup data is incorrect. Look at the lab 3 jupyter notebook. The target of the model was whether or not the given netflow was a DDOS attack. All other cases of malware were classified as "not a ddos attack." Review code. 
*	Data Preparation
    *	remove the TODO
    *	There is a section on app.py in here. This should be moved to the deployment section. This describes how data is prepared to be used against the trained model, while this "data preparation" section should describe preparing data for the fitting of the model.
    *	This section should not mention the modeling algorithm (RandomForest). Using RandomForst twice -- with different samples of data -- does not count as two different modeling algorithms. That's just one algorithm.
*	Modeling
    *	This section should include descriptions of about 3 different modeling algorithms that you tried, what library you used (sklearn, what version?) and any defaults that were changed for the algorithm functions you called.
    *	This is where you would describe a test-train split, and upsampling / downsampling. It is correct to leave onehotencoding in the Data Preparation section -- that counts as feature engineering. 
    *	Regarding your test-train split, how many rows were in each set? .5 split is a huge amount! How was the split done -- this is time-series data, did you do a stratified split, or time series split, or random, etc? Just describe what you did.
    *	In this section, you should describe your feature list. (You used all features. You describe this elsewhere, but you should describe it here instead.)
*	Evaluation
    *	This section needs a lot of work.
    *	Generally, figures need labels (e.g., Figure 1, Figure 2), so that they can be referenced in the text of the report.
    *	Figure 1
    *	I can't interpret the table of numbers in Figure 1. The column headers seem to not match the data below them. What is a "model score" -- is it accuracy? What is the 0 and 1 row? What threshold is this supposed confusion-matrix printed at?
    *	please report ratios in your colored confusion matrix, not counts. 
    *	The ROC plot should list one line for each of the three modeling algorithms used. See lab 3.
    *	A precision-recall plot should be provided, again, with one line for each algorithm.
    *	A table reporting evaluation metrics for your three algorithms should be provided. For example, if "accuracy" were your evaluation metric, you would report one row for each algorithm, and the accuracy of each. Same if you chose log-loss, or whatever.
    *	For the model that you choose, you should report a feature importance plot. What were the most predictive features? 
*	Deployment section
    *	seems to have some "project requirements" kruft.
    *	This is where you would talk about pickling the best-performing model from the previous section, and loading that pickle into the flask app.
    *	Oh, you need to describe that the app is written in flask. Perhaps link to the report repository for the flask app, in case you copy your report off elsewhere.
    *	Round off the threshold float please!
    *	Again, move the app.py "data preparation" description down to this section. Link to the argus and ra config files.
    *	This section should say what the use-case for the deployed website is. You may also choose to include screenshots of the website. I would not include a link to a live version of it in this report.
 
*	The elephant in the room is, what happened to KDDCup? It was promised in the introductory sections of the paper. Maybe do this: drop references to KDDCup from all beginning sections of the paper, except in the Deployment section, say that you also deployed a trained KDDCup model to the flask app. Perhaps say that the training of that model is described elsewhere, and link out to it.
    *	If you include descriptions of several modeling algorithms, then I would argue that this counts as including more than one model, especially since you have deployed models trained on two different datasets to the website.

**Comment by deargle** [comment]
Excellent revision! Stellar job on the modeling and evaluation sections. Relatively minor revision requests below, then I think this paper will be ⭐ 

# Business Understanding

Business understanding section incorrectly states that kddcup99 will be used for multiclass predictions, and that ctu-13 will be used to classify "malicious vs not." Technically incorrect, the ctu-13 can only classify whether or not the traffic is from a botnet.

* "The answer to this is an automated process that will follow up on alerts to determine if network traffic is from a botnet." incorrect, your system does not follow up on alerts. Rather, it _generates_ alerts for humans to follow up on. Add a sentence differentiating this from rule-based detection. "Unlike signature-based intrusion detection systems such as Snort, machine learning approaches such as ours can recognize new attacks based on probability of similarity to attacks seen in the past." Basically, just clarify that you're using machine learning to train a model.

# Data Preparation

furthermore, I remembered that you need to describe in your data preparation section that you feature-engineered the target variable to be anything "from-botnet-*" as 1 and all else as 0. And emphasize that that's your target. That's important! That says what you're predicting!

# Modeling
* Add a reference citation for SMOTE (a link will suffice)
* Clarify that your table of "features" is sample data. And you need to change the Label example in that example -- *wait, I hope you didn't include Label as a feature for training?!*

# Model Evaluation
* "we got average results with the Logistic Regression model." what does this mean? avoid use of word "average" in this statistical context if you don't actually mean average. Clarify.
* "Figures" in-text reference should be "Figures", not "figures."
* "We have shown" => "we show" present tense
* _please please_ only show normalization confusion matrix. Non-normalized not meaningful.
* What threshold was used for the predictions that the confusion matrices were built from? If you used `.predict()` instead of `.predict_proba`, then I think the threshold is .5. Specify.
* "In the table, "model score" refers to a scikit-learn function used which calculates the mean accuracy for the model." -- I would just rename in your tables and text to be to "mean accuracy." Or better, in the table, do "Model Score (Mean Accuracy)" Wait, what does "mean accuracy" even mean? Is it just "accuracy"?
* Random Forest did _not_ perform better than Gradient Boosting on "Model Score." I think that you are still justified in using random forest over gradient boosting though, since it does better on metrics that favor positive predictions (all your other metrics). And in an infosec business context, you could say that false positives are very time-wise costly.

# Deployment
* "The threshold set for deployment for CTU-13 was 0.25, which is the threshold that maximizes the F1 score for our CTU-13 model." clarify that this is a decision boundary threshold for predicting "from botnet" vs not.
* "click here" needs a dot at the end
* "while the Kddcup99 model will be able to predict whether it is a DDoS attack, or not, among 4 main classes." What does "4 main classes" refer to? I would just drop that reference. I think you're talking about the different kinds of ddos attacks in the kddcup99 dataset, but my target variable collapses among all those. Confusing to read.

**Comment by deargle** [comment]
:+1: :tada: 💯
